cli/commands/project_sync.go

```diff
@@ -65,33 +72,50 @@ func (cmd *ProjectSync) Commands() []cli.Command {
 		Usage:       "Stops a unison sync on local project directory. Optionally provide a volume name.",
 		ArgsUsage:   "[optional volume name]",
 		Description: "Volume name will be discovered in the following order: argument to this command > outrigger project config > docker-compose file > current directory name",
-		Before:      cmd.Before,
-		Action:      cmd.RunStop,
+		Flags: []cli.Flag{
+			// Override the local sync path.
+			cli.StringFlag{
+				Name:  "sync-path",
+				Value: "",
+				Usage: "Specify the location in the local filesystem to be synced. If not used it will look for the directory of project configuration or fall back to current working directory. Use '--sync-path=.' to guarantee current working directory is used.",
+			},
+		},
+		Before: cmd.Before,
+		Action: cmd.RunStop,
 	}
 
 	return []cli.Command{start, stop}
 }
 
-// Start the unison sync process
+// Start the Unison sync process.
 func (cmd *ProjectSync) RunStart(ctx *cli.Context) error {
 	cmd.Config = NewProjectConfig()
 	cmd.out.Verbose.Printf("Loaded project configuration from %s", cmd.Config.Path)
-	volumeName := cmd.GetVolumeName(ctx, cmd.Config)
+	// Determine the working directory for CWD-sensitive operations.
+	var workingDir string
+	if syncPath, err := cmd.DeriveLocalSyncPath(cmd.Config, ctx.String("sync-path")); err != nil {
```

Comment:
could you not use `workingDir` here as it is already defined as a var and avoid the else all together?

Reply:
I declare the var so it's in function instead of block scope, but the := operator is caught by the compiler as an overwrite. I could nest the whole function inside the if with reversed err check.